### PR TITLE
feat: Orchestrator Integration

### LINKED ISSUE #4390: Orchestrator UX integration
1. Provide new choice in recognizer (orchestrator) with the designed tooltip and help link. 
   * after user choose orchestrator the downloading will starting by default, before downloaded, the start bot is not allowed
   * before https://github.com/microsoft/botbuilder-dotnet/issues/4651 is addressed, after user choose orchestrator, we will have to user $designer.id to mark this is orchestrator instead of LUIS

2. Create a new filed in settings page for the base model selection (by default is the least layer bert-3layers)
3. Download the base model to a shared location. Consider both web scenario and electron scenario. 
4. Generate the recognizer dialogs once user choose Orchestrator, related PR: https://github.com/microsoft/BotFramework-Composer/pull/435
5. Do the orchestrator:build + luis:build based on the dialog choices (@lei9444)
6. Verify it works locally (at this time, remote deploy should not work, because base model missing in remote)

### PR DESCRIPTION
**Pull request by taicchoumsft**
## Description
This is the full **local** e2e flow for [Orchestrator](https://github.com/microsoft/BotBuilder-Samples/tree/main/experimental/orchestrator)

What is here:
1) Orchestrator behind feature flag
1) Orchestrator schema, recognizer flow, and UI improvements by @lei9444 (thanks Leilei)
1) Orchestrator packaging tested on Electron Windows 10, Mac (10.15.7), Linux (Ubuntu 20.04) from CI builds
1) Orchestrator e2e **local** publish flow on the C# WebApp runtime only

What isn't:
1) Remote publishing flows
2) Ability for runtime to independently manage model (#4215)
3) Runtime support for anything but the C# WebApp runtime

## Task Item
Closes #4390 , #4209

## Screenshots
image
image

### CODE REVIEW

**Review by tsuwandy — APPROVED** [review]

**Review by cwhitten — APPROVED** [review]

File: Composer/packages/server/src/models/bot/builder.ts [hunk 1]
```diff
@@ -101,6 +109,73 @@ export class Builder {
...
+    // build snapshots from LU files
+    const returnData = await this.orchestratorBuilder(luFiles, modelPath);
+
+    // write snapshot data into /generated folder
+    const snapshots: any = {};
+    for (const dialog of returnData.outputs) {
+      const bluFilePath = Path.resolve(this.generatedFolderPath, dialog.id.replace('.lu', '.blu'));
+      snapshots[dialog.id.replace('.lu', '').replace(/[-.]/g, '_')] = bluFilePath;
+
+      writeFile(bluFilePath, Buffer.from(dialog.snapshot), (err) => {
```
Comment (hatpick) [inline Composer/packages/server/src/models/bot/builder.ts:155]: why not using the "promisified" writeFile from fs-extra and await?

File: Composer/packages/server/src/models/bot/builder.ts [hunk 2]
```diff
@@ -101,6 +109,73 @@ export class Builder {
...
+    const orchestratorSettings: any = {
+      orchestrator: {
+        ModelPath: modelPath,
+      },
+    };
+
+    orchestratorSettings.orchestrator.snapshots = snapshots;
+    const orchestratorSettingsPath = Path.resolve(this.generatedFolderPath, 'orchestrator.settings.json');
+
+    writeFile(orchestratorSettingsPath, JSON.stringify(orchestratorSettings), (err) => {
```
Comment (hatpick) [inline Composer/packages/server/src/models/bot/builder.ts:172]: same here

File: Composer/packages/server/src/models/bot/builder.ts (hunk 1 above)
Comment (taicchoumsft) [reply Composer/packages/server/src/models/bot/builder.ts]: Thanks @hatpick, I completely missed this.  Fixed this and the `pathExists` function on line 138.

File: Composer/packages/server/src/models/bot/builder.ts (hunk 2 above)
Comment (taicchoumsft) [reply Composer/packages/server/src/models/bot/builder.ts]: done.

**Review by hatpick — APPROVED** [review]

**Pull request merged**